src/core.jl

```diff
@@ -1594,13 +1586,12 @@ function gradients(y, x::AbstractArray, grad_y=nothing)
     return out
 end
 
-gradients(y, x, grad_y=nothing) = gradients(y, [x], grad_y)[1]
-
-function c_gradients(y, x, dx=nothing)
+c_gradients(y, x, dx=nothing) = c_gradients([y], x, dx)
+function c_gradients(y::AbstractArray, x::AbstractArray, dx=nothing)
     status = Status()
     graph = get_def_graph()
     grads = Array{Port}(length(x))
-    if dx === nothing
+    if dx isa AbstractArray{Void} || dx isa Void
```

Comment:
Should we use `<:` instead of `isa` here?

Reply:
I checked this with `@code_llvm` (well a MWE).
Either way it is removed during specialization.
(as is `==nothing`)

I think `isa` is cleanest